### PR TITLE
[WIP] Make title and favicon configurable

### PR DESCRIPTION
**Pull request by nchevobbe**
Also changed the launchpad default favicon (to the unicode rocket emoji).
This works for the title, but not for the favicon yet.

I don't know jack about Express so I just hacked a little to see how I can done that.
I was hoping to declare the containing folder of the favicon as a static assets so I can use the favicon name directly but I face some problems:

1. I don't see how I can declare a relative path to the favicon in the `development.json` file (in Reps for example), since express uses the `devtools-launchpad` path.
2. Even if I use an absolute path in the config, I can't get the favicon to be displayed. The weird thing is that I can see it in `localhost:8000/favicon.png` though :confused: 

I'll dig this more, just wanted to know if there were better strategies here.

Also this would conflict with #44 , so I'm waiting it to be merged so I can rebase on top of it.

### CODE REVIEW

File: packages/devtools-launchpad/bin/development-server.js [hunk 1]
```diff
@@ -35,7 +35,13 @@ function httpGet(url, onResponse) {
 function serveRoot(req, res) {
   const tplPath = path.join(__dirname, "../index.html");
   const tplFile = fs.readFileSync(tplPath, "utf8");
-  res.send(Mustache.render(tplFile, { isDevelopment: isDevelopment() }));
+  res.send(Mustache.render(tplFile, {
+    isDevelopment: isDevelopment(),
+    pageTitle: getValue("pageTitle") || getValue("title"),
```
Comment (jasonLaster) [inline packages/devtools-launchpad/bin/development-server.js]: Why two titles?

File: packages/devtools-launchpad/bin/development-server.js [hunk 2]
```diff
@@ -35,7 +35,13 @@ function httpGet(url, onResponse) {
...
+    favicon: getValue("assets.favicon")
+      ? path.basename(getValue("assets.favicon"))
+      : "launchpad-favicon.png"
```
Comment (jasonLaster) [inline packages/devtools-launchpad/bin/development-server.js]: Hmm not sure about this one. Perhaps tools add their favs in the same launchpad dir :)

File: packages/devtools-launchpad/bin/development-server.js [hunk 3]
```diff
@@ -91,6 +97,14 @@ function startDevServer(devConfig, webpackConfig) {
   const app = express();
   app.use(express.static("assets/build"));
   app.use(express.static(path.join(__dirname, '../assets')));
+
+  let assets = getValue("assets");
+  if (assets) {
+    Object.keys(assets).forEach(key => {
+      app.use(express.static(path.dirname( assets[key])));
+    });
```
Comment (jasonLaster) [inline packages/devtools-launchpad/bin/development-server.js]: Hmm. What does this do?

File: packages/devtools-launchpad/bin/development-server.js (hunk 2 above)
Comment (juliandescottes) [inline packages/devtools-launchpad/bin/development-server.js]: Or we keep href="favicon.png" in index.html and the tools just define custom routes to point to their local favicon ?

**Comment by nchevobbe** [comment]
@juliandescottes makes some good points in https://github.com/devtools-html/devtools-core/issues/27#issuecomment-269775976 .

File: packages/devtools-launchpad/bin/development-server.js (hunk 2 above)
Comment (jasonLaster) [inline packages/devtools-launchpad/bin/development-server.js]: yea - that's fair

**Comment by jasonLaster** [comment]
Hmm, that *is* a good point. I think I prefer adding more config variables for the template. Either way, that probably helps us as we're treating the index as a mustache template.

This stuff is never great :P

**Comment by jasonLaster** [comment]
ping @nchevobbe lets try and get this in soon.

**Comment by nchevobbe** [comment]
@jasonLaster sure, thanks for the reminder

**Comment by nchevobbe** [comment]
@juliandescottes @jasonLaster I just pushed an update to this PR. Since I wanted the favicon to be fully configurable (and not dependant of where the devtools-launchpad modules is in the tree-dependency), I added a `rootDir` argument to the `startDevServer` function, so we can access it in devtools-launchpad. Which makes possible to declare in the configuration file a path to a favicon, wherever it is in the repo.

I successfully tested it in devtools-rep (with yarn link), with and without favicon configuration specified.

Here's what the configuration looks like : 
```json
  "title": "Firefox Devtools Reps",
  "favicon": "src/launchpad/assets/reps-favicon.png",
```

Feel free to tell me if you think that it's a bad idea :) I'm not used to things like that.

**Pull request merged**